[16.0][MIG] stock_picking_batch_extended: Migration to 16.0

BorisFL opened the pull request.

Commit:
[FIX] stock_picking_mass_action return backorder if any of pickings need it

AdriaGForgeFlow committed:
[12.0][FIX] stock_picking_batch_extended

pedrobaeza committed:
[MIG] stock_picking_batch_extended: Migration scripts

- Converts old stock.batch.picking in stock.picking.batch
- Restore link in stock.picking
- Other needed stuff

carlosdauden committed:
[FIX] stock_picking_batch_extended: float_is_zero required parameters

carlosdauden committed:
[IMP] stock_picking_batch_extended: UX improvements

Commit:
[IMP] add post-init hook to apply settings

carlosdauden committed:
[IMP] stock_picking_batch_extended: Add followers, attachments, chatter and activities

sergio-teruel committed:
[IMP] stock_picking_batch_extended: Request confirmation to cancel pickings

pedrobaeza committed:
[IMP] stock_picking_batch_extended: black, isort, prettier

sergio-teruel committed:
[MIG] stock_picking_batch_extended: Migration to 13.0

* Standard procedure
* Fix tests

sergio-teruel committed:
[IMP] stock_picking_batch_extended: Add kanban view in batch action for small screens.

OCA-git-bot committed:
stock_picking_batch_extended 13.0.1.1.0

carlosdauden committed:
[IMP] stock_picking_batch_extended: Add optional picking_count column in tree view
TT28975

OCA-git-bot committed:
stock_picking_batch_extended 13.0.1.2.0

victoralmau committed:
[IMP] stock_picking_batch_extended: Add confirm to validate and done buttons.
TT30796

OCA-git-bot committed:
stock_picking_batch_extended 13.0.1.3.0

Commit:
[IMP] stock_picking_batch_extended: black, isort, prettier

Commit:
[MIG] [14.0] stock_picking_batch_extended: Migration to 14.0

cesar-tecnativa committed:
[MIG] stock_picking_batch_extended: Migration to 15.0

sergio-teruel committed:
[IMP] stock_picking_batch_extended: Refactor. Odoo has improved batch pickings

Commit:
[UPD] Update stock_picking_batch_extended.pot

OCA-git-bot committed:
[UPD] README.rst

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-workflow-15.0/stock-logistics-workflow-15.0-stock_picking_batch_extended
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-workflow-15-0/stock-logistics-workflow-15-0-stock_picking_batch_extended/

BorisFL commented:
[[MIG] migration stock_picking_batch_extended from 15.0 to 16.0](https://github.com/OCA/stock-logistics-workflow/pull/1236/commits/15feeec3720de19b8d3c6373e851f8a0c9f940c0)

RodrigoBM commented:
https://github.com/OCA/stock-logistics-workflow/issues/1101

RodrigoBM commented:
@rousseldenis can you add to issue https://github.com/OCA/stock-logistics-workflow/issues/1101

rousseldenis commented:
/ocabot migration stock_picking_batch_extended

rousseldenis commented:
@BorisFL Could you check this https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-16.0?

Could you follow especially Full process part ? Thanks

BorisFL commented:
> @BorisFL Could you check this https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-16.0?
> 
> Could you follow especially Full process part ? Thanks

Sure! I just uploaded the corrections. Can you review them to make sure they are correct?

rousseldenis commented on .gitignore:
```diff
@@ -72,3 +72,5 @@ docs/_build/
 # OCA rules
 !static/lib/
+
+setup/stock_picking_batch_extended/
```
You should not include this

rousseldenis commented on stock_picking_batch_extended/__manifest__.py:
```diff
@@ -1,14 +1,12 @@
-    "version": "15.0.2.0.0",
-    "author": "Camptocamp, " "Tecnativa, " "Odoo Community Association (OCA)",
-    "development_status": "Mature",
-    "maintainers": ["gurneyalex", "carlosdauden", "i-vyshnevska"],
```
You cannot remove existing maintainers unless they accepted it

rousseldenis commented on stock_picking_batch_extended/__manifest__.py:
```diff
@@ -1,14 +1,12 @@
     "name": "Stock batch picking extended",
     "summary": "Allows manage a lot of pickings in batch",
-    "version": "15.0.2.0.0",
-    "author": "Camptocamp, " "Tecnativa, " "Odoo Community Association (OCA)",
```
You cannot remove original authors

rousseldenis commented on stock_picking_batch_extended/__manifest__.py:
```diff
@@ -1,14 +1,12 @@
     "summary": "Allows manage a lot of pickings in batch",
-    "version": "15.0.2.0.0",
-    "author": "Camptocamp, " "Tecnativa, " "Odoo Community Association (OCA)",
-    "development_status": "Mature",
```
Let the development status

rousseldenis commented on stock_picking_batch_extended/__manifest__.py:
```diff
@@ -19,10 +17,12 @@
     ],
-    "installable": True,
-    "post_init_hook": "post_init_hook",
+    "application": False,
```
Those two values are the default, remove them

rousseldenis reviewed (changes requested):
@BorisFL Thanks for this.

Unfortunately, you cannot change the authorship everywhere. Please keep them. You can add yourself in code parts you improve only.

Could you also limit your migration changes in migration commit, then, do other commit(s) for improvements ?

Thanks

BorisFL commented:
@rousseldenis  I have already fixed the bugs and sorry for deleting the other authors.

mymage committed:
Translated using Weblate (Italian)

Currently translated at 21.4% (15 of 70 strings)

Translation: stock-logistics-workflow-15.0/stock-logistics-workflow-15.0-stock_picking_batch_extended
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-workflow-15-0/stock-logistics-workflow-15-0-stock_picking_batch_extended/it/

sysadminmatmoz committed:
Translated using Weblate (Slovenian)

Currently translated at 87.1% (61 of 70 strings)

Translation: stock-logistics-workflow-15.0/stock-logistics-workflow-15.0-stock_picking_batch_extended
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-workflow-15-0/stock-logistics-workflow-15-0-stock_picking_batch_extended/sl/

sergio-teruel committed:
[IMP] stock_picking_batch_extended: Display standard picking tree view in batch pickings

carlosdauden committed:
[FIX] stock_picking_batch_extended: Administration settings required. Use _for_xml_id instad of read

OCA-git-bot committed:
stock_picking_batch_extended 15.0.2.0.1

pedrobaeza commented:
Please include #1276 and #1277

OCA-git-bot committed:
stock_picking_batch_extended 15.0.2.1.0

sergio-teruel committed:
[IMP] stock_picking_batch_extended: Display batch picking in form view

OCA-git-bot committed:
stock_picking_batch_extended 15.0.2.2.0

BorisFL committed:
[IMP] stock_picking_batch_extended: pre-commit stuff

BorisFL commented:
@pedrobaeza , I have added the issues [#1276](https://github.com/OCA/stock-logistics-workflow/pull/1276) and [#1277](https://github.com/OCA/stock-logistics-workflow/pull/1277).

hugo-cordoba reviewed (commented):
The development looks correct, but leave the names of the previous contributors please. Add your name as a contributor in the CONTRIBUTORS.rst file along with the rest of the previous developers.

pedrobaeza commented on stock_picking_batch_extended/__manifest__.py:
```diff
@@ -0,0 +1,32 @@
+    "version": "16.0.1.0.0",
+    "author": "Camptocamp, "
+    "Tecnativa, "
+    "FactorLibre, "
```
This migration doesn't qualify your company for being co-author. A significant contribution to the module should be done for that elevated thing. You are already in contributors section, so it's enough.

BorisFL replied:
I'm sorry, I just fixed it.

pedrobaeza commented on stock_picking_batch_extended/__manifest__.py:
```diff
@@ -0,0 +1,32 @@
+    "FactorLibre, "
+    "Odoo Community Association (OCA)",
+    "development_status": "Mature",
+    "maintainers": ["gurneyalex", "carlosdauden", "i-vyshnevska", "BorisFL"],
```
Don't put yourself as maintainer if you don't plan to maintain the module.

BorisFL replied:
I just fixed it.

pedrobaeza commented on stock_picking_batch_extended/__manifest__.py:
```diff
@@ -20,10 +21,9 @@
         "wizard/stock_picking_to_batch_views.xml",
     ],
-    "installable": True,
-    "post_init_hook": "post_init_hook",
```
Why removing this?

pedrobaeza replied:
This comment is pending

rousseldenis replied:
@BorisFL

rousseldenis replied:
@BorisFL Could you restore post_init_hook ? Thanks

BorisFL replied:
Yes, I have just recovered it

pedrobaeza commented on stock_picking_batch_extended/views/report_batch_picking.xml:
```diff
@@ -13,22 +13,21 @@
-                        <div class="col-3">
-                            <strong>Date:</strong>
-                            <span t-esc="now()" t-options="{'widget': 'datetime'}" />
+            <t t-foreach="docs" t-as="doc">
```
Don't add this loop here. Do it in another template and passing `t-language` for each document. See the scheme in other standard documents.

BorisFL replied:
Changes applied, thanks for the notation

pedrobaeza commented on stock_picking_batch_extended/views/res_config_settings_views.xml:
```diff
@@ -5,25 +5,27 @@
-                            <label for="use_oca_batch_validation" />
-                            <div class="text-muted">
-                                Apply OCA batch validation
+            <data>
```
This `<data>` label is not correct inside here and not needed.

BorisFL replied:
Fixed

pedrobaeza commented on stock_picking_batch_extended/wizard/stock_picking_to_batch.py:
```diff
@@ -1,4 +1,5 @@
 # Copyright 2012-2016 Camptocamp SA
+# Copyright 2023 FactorLibre - Boris Alias
```
No change in this file, so remove this copyright

BorisFL replied:
It has already been deleted

pedrobaeza commented on stock_picking_batch_extended/views/report_batch_picking.xml:
```diff
@@ -14,128 +14,124 @@
-                            <strong>Printed by:</strong>
-                            <span t-field="user.name" />
-                        </div>
+    <template id="report_batch_picking_document">
```
Why this has a different level of indentation?

BorisFL replied:
I don't understand what it refers to. The template has the same level of indentation as the previous record `action_report_batch_picking` or the following template `report_batch_picking`

If you are referring to the report_name or report_file fields, those are line breaks introduced by pre-commit.

pedrobaeza commented on stock_picking_batch_extended/readme/CREDITS.rst:
```diff
@@ -1 +1 @@
-The migration of this module from 13.0 to 14.0 was financially supported by Camptocamp
+The migration of this module from 15.0 to 16.0 was financially supported by FactorLibre
```
Please remove this file.

BorisFL replied:
Fixed

pedrobaeza commented on stock_picking_batch_extended/models/stock_warehouse.py:
```diff
@@ -1,5 +1,6 @@
 # Copyright 2016 Camptocamp SA
-# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+# Copyright 2023 FactorLibre - Boris Alias
```
No change should be made here

BorisFL replied:
Fixed

rousseldenis commented on stock_picking_batch_extended/data/batch_picking_actions_server.xml:
```diff
@@ -1,6 +1,4 @@
 <?xml version="1.0" encoding="utf-8" ?>
-<!-- Copyright 2018 Tecnativa - Carlos Dauden
```
@BorisFL Why removing this ?

BorisFL replied:
I will fix it, I'm sorry

rousseldenis replied:
@BorisFL 
Could you restore it as it was before:
![image](https://github.com/OCA/stock-logistics-workflow/assets/19529533/7db3b7aa-088a-48c7-bfd6-384f5b524e55)

BorisFL replied:
Fixed

rousseldenis commented on stock_picking_batch_extended/models/stock_move.py:
```diff
@@ -0,0 +1,16 @@
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    has_move_lines = fields.Boolean(compute="_compute_has_move_lines")
```
This deserves a separate module as it is a generic feature and not specific to batch extended

BorisFL replied:
I have generated another module for that

rousseldenis commented on stock_picking_batch_extended/models/stock_move_line.py:
```diff
@@ -0,0 +1,17 @@
+class StockMoveLine(models.Model):
+    _inherit = "stock.move.line"
+
+    product_uom_qty = fields.Float(
```
What does this field represent ?

rousseldenis replied:
I think it is not used anymore as replaced by reserved_qty field. Remove it

BorisFL replied:
I'll remove it

rousseldenis commented:
@BorisFL Could you rebase and fix pre-commit ?

stefan-tecnativa commented on stock_move_compute_move_lines/readme/DESCRIPTION.rst:
```diff
@@ -0,0 +1,38 @@
+This module is based on `picking_dispatch <https://github.com/OCA/stock-logistics-workflow/tree/8.0/picking_dispatch>`_
```
This README is the one of the migration module, please modify it to explain what this new module does.

BorisFL replied:
I have modified the README of this module.

stefan-tecnativa commented on stock_picking_batch_extended/models/product_product.py:
```diff
@@ -0,0 +1,11 @@
+# Copyright 2012-2014 Alexandre Fayolle, Camptocamp SA
+# Copyright 2023 FactorLibre - Boris Alias
```
Review please the files that you haven't modify but added this.

BorisFL replied:
Fixed

stefan-tecnativa commented on stock_picking_batch_extended/wizard/stock_picking_to_batch_views.xml:
```diff
@@ -0,0 +1,50 @@
+                    </field>
+                </group>
+            </xpath>
+            <xpath expr="//button[@name='attach_pickings']" position="after">
+                <button
+                    name="action_create_batch_with_mode_existing"
+                    string="Add to existing batch"
+                    attrs="{'invisible': [('mode', '=', 'create')]}"
+                />
+            </xpath>
```
This is not having effect. What was the intention here?

BorisFL replied:
Sorry. The values that the "mode" field now has are "existing" or "new". In case the option to create a new grouping is chosen, the button to add the selected delivery orders to an existing grouping disappears. Now it functions correctly.

sergio-teruel commented:
Why do you add the picking file?? This operatios already done by Odoo 
This PR , I think that is not ok

sergio-teruel reviewed (changes requested):
Please remove all extra actions from versions older than 15.0

BorisFL commented:
Hello, I am reviewing the modifications

BorisFL commented:
> I rebase mi commits

BorisFL commented:
@rousseldenis I rebase mi commits

rousseldenis commented on stock_move_compute_move_lines/__manifest__.py:
```diff
@@ -0,0 +1,18 @@
+# Copyright 2012-2014 Alexandre Fayolle, Camptocamp SA
```
@BorisFL This module seems to not be part of stock_picking_batch_extended migration

BorisFL replied:
In a previous [comment](https://github.com/OCA/stock-logistics-workflow/pull/1236#discussion_r1180394498), you asked me to split that functionality into another module. Should I directly delete that module in this commit?

sergio-teruel replied:
Please... remove this module from this commit

BorisFL replied:
I just deleted the module

BorisFL committed:
[MIG] migration stock_picking_batch_extended from 15.0 to 16.0

stefan-tecnativa commented on stock_picking_batch_extended/migrations/15.0.1.0/pre-migration.py:
You must remove any migration scripts from previous versions https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-16.0

BorisFL replied:
It's already fixed

BorisFL commented:
@stefan-tecnativa , I have made the mentioned modifications. Can you review them?

rousseldenis commented on stock_picking_batch_extended/tests/test_batch.py:
```diff
@@ -10,9 +11,9 @@ def _create_product(self, name, product_type="consu"):
         return self.env["product.product"].create({"name": name, "type": product_type})
 
     def setUp(self):
```
@BorisFL Could you change this to classmethod ?

BorisFL replied:
I have converted it to a classmethod

rousseldenis commented on stock_picking_batch_extended/tests/test_batch.py:
```diff
@@ -266,7 +255,7 @@ def test_package(self):
         ]
         group.run(procurements)
-        pickings = self.picking_model.search([("group_id", "=", group.id)])
+        pickings = self.env["stock.picking"].search([("group_id", "=", group.id)])
```
Why changing this ?

BorisFL replied:
It was a mistake, i fixed it

rousseldenis commented on stock_picking_batch_extended/tests/test_batch.py:
```diff
@@ -293,10 +282,10 @@ def test_remove_undone(self):
         self.assertEqual("cancel", self.batch.state)
-        self.assertEqual(1, len(self.batch.picking_ids))
-        self.assertEqual(self.picking2, self.batch.picking_ids)
+        self.assertEqual(0, len(self.batch.picking_ids))
```
Why this test has changed ?

BorisFL replied:
This change is due to the fact that in version 16 of Odoo base, when you cancel a delivery order, it is removed from its grouping if it had one. This can be found on line 22 of the stock_move file in the stock_picking_batch module

![imagen](https://github.com/OCA/stock-logistics-workflow/assets/122611176/2a66988f-20cf-4ea0-99a0-e6b5bd4e2633)

rousseldenis commented:
@BorisFL Some questions or changes are still blocking this

sergio-teruel commented:
Hy @BorisFL Why do you recover the stock_picking file from older versions??
In v15 the actions defined on picking are not needed...
Why now do you need it for v16?

BorisFL commented:
@sergio-teruel I accidentally recovered the file. Initially, I was migrating from version 14, and that's why the file got recovered. I have just deleted it now.

BorisFL commented:
Hi @rousseldenis . I made the corrections you asked for. Please take a look at them when you can.

rousseldenis commented on stock_picking_batch_extended/data/batch_picking_actions_server.xml:
```diff
@@ -10,11 +10,11 @@
             name="binding_model_id"
             ref="stock_picking_batch.model_stock_picking_batch"
         />
+        <field name="type">ir.actions.server</field>
```
No need to add this as this is the default.

BorisFL replied:
Fixed

rousseldenis commented on stock_picking_batch_extended/data/batch_picking_actions_server.xml:
```diff
@@ -26,10 +26,9 @@ action = env.ref('stock.action_report_picking').report_action(pickings)
-action = env.ref('stock.action_report_delivery').report_action(pickings)
+        <field name="code">
+pickings = records.mapped('picking_ids').filtered(lambda p: p.state != 'cancel')
+action = env['ir.actions.report'].sudo()._get_report_from_name('stock.report_delivery').report_action(pickings)
```
@BorisFL Are you sure about the sudo ? As it is done in _get_report_from_name()

BorisFL replied:
fixed

rousseldenis commented on stock_picking_batch_extended/hooks.py:
```diff
@@ -6,5 +6,5 @@
 def post_init_hook(cr, registry):
     env = api.Environment(cr, SUPERUSER_ID, {})
-
-    env["res.company"].search([]).write({"use_oca_batch_validation": True})
```
Why changing this?

rousseldenis commented on stock_picking_batch_extended/models/stock_batch_picking.py:
```diff
@@ -99,6 +105,22 @@ def action_print_picking(self):
         self.mapped("active_picking_ids").write({"batch_id": False})
+        self.verify_state()
+
+    def verify_state(self):
```
Last but not least, this is an addition in behavior from previous version.

IMHO, this module deserves a good DESCRIPTION.rst in readme folder to explain what it does as 'extended' is too general term to understand at first sight. Thanks

BorisFL replied:
I have fixed it. I have cleaned up some errors from an initial import I made from v14, and the only change is the removal of "remove_undone_pickings." This change ensures that when a picking is canceled, its grouping is removed in v16

rousseldenis replied:
Ok, nevertheless, a good module description could be a great improvement :smiley:

BorisFL replied:
I have added a description to the readme and index files

FrancoMaxime commented on stock_picking_batch_extended/__manifest__.py:
```diff
@@ -1,11 +1,12 @@
     "summary": "Allows manage a lot of pickings in batch",
-    "version": "15.0.2.2.0",
+    "version": "16.0.1.0.0",
     "author": "Camptocamp, " "Tecnativa, " "Odoo Community Association (OCA)",
```
Hi @BorisFL , thanks for contributing.
can you fix this old pre-commit result?
```suggestion
    "author": "Camptocamp, Tecnativa, Odoo Community Association (OCA)",
```

BorisFL replied:
I have fixed the concatenation of the authors

FrancoMaxime replied:
thanks :)

FrancoMaxime reviewed (approved):
LGTM, minor suggestion

rousseldenis commented:
@pedrobaeza @sergio-teruel Could you update your review ?

BorisFL commented:
Hello  @pedrobaeza and @sergio-teruel .

I have made the mentioned modifications. Can you review them?

BorisFL commented:
Good morning @pedrobaeza and @sergio-teruel , I apologize for insisting, but have you received any notification? Can you review your comments?

xAdrianCif commented:
Hello, @sergio-teruel  Could you review the changes made and leave an approval if you think it's appropriate? If you see the need for any additional changes, please let me know. From now on, I will take care of this PR, so if anyone thinks we should make any changes, please let me know as well! Thanks to all!

xAdrianCif commented:
Hello everyone, the person who developed and maintained this PR is no longer collaborating with our company. We have detect some minor issues so we have decided to close it, reanalyze it, and develop it again. We will upload the new one as soon as possible.

PD: I do not have the permissions to close this PR. If someone could do it, I would greatly appreciate it.

CC @pedrobaeza @rousseldenis @sergio-teruel @FrancoMaxime

nicomacr commented:
@xAdrianC-FactorLibre any news about the new PR of the module?

xAdrianCif commented:
Hello @nicomacr , we have conducted a preliminary analysis and believe that changes need to be made compared to the previous MR. However, I cannot provide much information regarding actions or possible dates as we are currently focused on other priorities.

xAdrianCif commented:
@nicomacr @rousseldenis @FrancoMaxime 
Hello everyone, a colleague has carried out the migration that was postponed in this issue. I'll leave the link here in case you're still interested. Thank you!
https://github.com/OCA/stock-logistics-workflow/pull/1353

nicomacr commented:
@xAdrianCif Thank you for taking care of the migration of the module.
